arraycontext/container/traversal.py

```diff
@@ -744,7 +744,11 @@ def _unflatten(template_subary: ArrayOrContainerT) -> ArrayOrContainerT:
                     "'template' is too large")
 
             if strict:
-                if template_subary.dtype != ary.dtype:
+                # Checking strides for 0-sized arrays is ill-defined in numpy
+                if (
+                    (template_subary.strides != ary.strides)
+                    and (template_subary.size != 0)
+                ):
```

Comment:
Not the correct spot to insert this check. Would've been better to directly set the source branch as the branch in https://github.com/kaushikcfd/arraycontext/pull/4 :).

Reply:
This is fixed in the latest version. Thanks!